### PR TITLE
Install Linkage Checker for spring-cloud-gcp-dependencies BOM

### PR DESCRIPTION
**Pull request by suztomo**
- Install Linkage Checker for spring-cloud-gcp-dependencies BOM
- run `mvn verify` as a separate task in Travis, because Linkage Checker cannot handle multi-thread Maven execution with `install` task.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,8 +31,10 @@ script:
   - if [ "$INTEGRATION_TEST_FLAGS" != "" ]; then
       ./mvnw test -B -P codecov ${INTEGRATION_TEST_FLAGS} && bash <(curl -s https://codecov.io/bash) -F integration;
     fi;
+  # run enforcer rules
+  - ./mvnw -T 1.5C verify -DskipTests=true -Dspring.profiles.active=spring -B -V -P full-checkstyle
 install:
-  - ./mvnw -T 1.5C install -DskipTests=true -Dmaven.javadoc.skip=true -Dspring.profiles.active=spring -P full-checkstyle -B -V
+  - ./mvnw -T 1.5C install -DskipTests=true -Dmaven.javadoc.skip=true -Dspring.profiles.active=spring -Denforcer.skip=true -B -V
```
Comment (suztomo) [inline .travis.yml:37]: Running enforcer rule is part of the script 2 lines above.

**Comment by suztomo** [comment]
@meltsufin @elefeint PTAL.

The overhead of Travis execution time:

Before this PR. Travis's install step took 86 seconds https://travis-ci.org/github/spring-cloud/spring-cloud-gcp/builds/717023299 (16 min 54 sec)

Screen Shot 2020-08-17 at 12 01 40

After this PR, Travis's install step takes 349 seconds (291.54 + 57.54) https://travis-ci.org/github/spring-cloud/spring-cloud-gcp/builds/718674385 (20 min 3 sec)

Screen Shot 2020-08-17 at 12 03 59

**Comment by elefeint** [comment]
Perhaps we could have a daily linkage checker job that's outside of the main build process? Travis worked hard optimizing the build for multithreading...

**Comment by suztomo** [comment]
> Perhaps we could have a daily linkage checker job that's outside of the main build process?

That's a nice idea. Do you have a similar job I can replicate?

**Comment by meltsufin** [comment]
@ttomsu I would suggest you re-open this PR in our [new repo](https://github.com/GoogleCloudPlatform/spring-cloud-gcp) for the project, where we have granular builds using GitHub actions. This way you can have a build that only does the linkage checker.

**Comment by suztomo** [comment]
@meltsufin Good. I'll follow that.

**Comment by suztomo** [comment]
It hit another bug on Linkage Checker https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1609

**Pull request closed without merging**